integrations/gcp/gcp_core/feed_event.py

```diff
@@ -0,0 +1,68 @@
+import base64
+import json
+from typing import Any, NotRequired, TypedDict
+
+from port_ocean.core.ocean_types import RAW_ITEM
+
+from .search.searches import (
+    get_folder,
+    get_organization,
+    get_project,
+    get_topic,
+    search_resource,
+)
+from .search.utils import EXTRA_PROJECT_FIELD, AssetTypesWithSpecialHandling
+
+
+class FeedEvent(TypedDict):
+    message_id: str
+    asset_name: str
+    asset_type: str
+    project_id: NotRequired[str]
+    data: dict[Any, Any]
+
+
+class GotFeedCreatedSuccessfullyMessage(Exception):
+    pass
+
+
+class AssetHasNoProjectAncestorError(Exception):
+    pass
+
+
+def get_project_from_ancestors(ancestors: list[str]) -> str:
+    for ancestor in ancestors:
+        if ancestor.startswith("projects/"):
+            return ancestor
+    raise AssetHasNoProjectAncestorError
```

Comment:
why would this happen?

Reply:
There are organization level objects which im not currently digesting, maybe for future version we'll decide to add them after customers would request it

Reply:
wont it fail the resync?